### ISSUE TITLE
Add support for multiple examples

### ISSUE THREAD

**Issue by Elfocrash** [mattfrear/Swashbuckle.AspNetCore.Filters#109]
Since Swagger 2.0 and Open API 3.0 support multiple examples instead of just a single, it would be cool to add a `IMultipleExamplesProvider<RequestExample<T>>` interface to make use of the multiple examples support capabilities.

Here is an example of how they look in OpenAPI 3.0:

![image](https://user-images.githubusercontent.com/8199968/63273564-1d104800-c296-11e9-8bbd-4d9648a7042d.png)

![image](https://user-images.githubusercontent.com/8199968/63273037-1b925000-c295-11e9-94d9-23c403ec170a.png)

You need some sort of identifier for the example and the T object so you would need that's why the `RequestExample` wrapper class would look something like this:

```
public class RequestExample<T>
{
    public string Name { get; set; }
    public T Example { get; set; }
}
```

Then you just return an enumerable of those examples in the example provider and you can choose from multiple examples instead of a single one. This can be quite handy in different example scenarios.

Looking at the current implementation we would need to set the `Examples` property instead of the `Example` one which is currently being set.

![image](https://user-images.githubusercontent.com/8199968/63275517-8c3b6b80-c299-11e9-8174-a6ba88fd3f1a.png)

The `Name` value I'm talking about is the `Summary` in the `OpenApiExample` and the `Value` is just the `IOpenApiAny` which is used atm.

**Comment by mattfrear** [comment]
Hi

This sounds like a great idea, and I didn't know that Swagger supported that, so thank you for raising.

However, as per #108, I'd like to deprecate my examplesoperationfilter in favour of using the XML documentation already built in to Swashbuckle. Perhaps multiple examples could be added to the XML documentation:
```csharp
     /// <summary>
     /// The id of the person
     /// </summary>
     /// <example name="Jessica">10</example>
     /// <example name="Ron">11</example>
     public int Id { get; set; }

     /// <summary>
     /// The name of the person
     /// </summary>
     /// <example name="Jessica">Jessica Smith</example>
     /// <example name="Ron">Ron Stewart</example>
     public string Name { get; set; }
```

**Comment by tomkludy** [comment]
I submitted pull request #111 to add support for multiple examples.

**Comment by mattfrear** [comment]
Still waiting on unit tests to be added to that PR before I'll accept it. I might do this over the Christmas holidays if I get time.

**Comment by sabbadino** [comment]
I'd love to have this feature available

**Comment by ddeisadzevp** [comment]
Any updates to this?

**Comment by mattfrear** [comment]
I think the code is done by tomkludy, but I haven't tried using it. I'm unwilling to merge to master until there are unit tests added to the PR, ideally by tomkludy.

I was hoping to find some time to look at this over the Christmas break, but I was too busy. So I'm not sure when it will get picked up.

**Comment by ddeisadzevp** [comment]
Ok sounds good, thanks. @tomkludy do you need help, I can hop on adding unit tests if you haven't already done so locally but lmk.

**Comment by tomkludy** [comment]
Very sorry, I was hoping to get some time to add unit tests but I haven’t
been able to.  I would welcome help in this!

**Comment by mattfrear** [comment]
Thanks to @tomkludy and @pozy. 

Released https://www.nuget.org/packages/Swashbuckle.AspNetCore.Filters/5.1.0

**Comment by mattfrear** [comment]
@claorden lists are supported in XML examples. There is an example documented here: https://github.com/domaindrivendev/Swashbuckle.AspNetCore#include-descriptions-from-xml-comments

```csharp
    /// <summary>
    /// The sizes the product is available in
    /// </summary>
    /// <example>["Small", "Medium", "Large"]</example>
    public List<string> Sizes { get; set; }
```

**Comment by mattfrear** [comment]
Hi @claorden 

I'm pretty sure multiple examples are not supported with XML Comments, so you'll have to use my Filters package for now.

Matt
